src/main/kotlin/org/opensearch/commons/alerting/model/AlertContext.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.commons.alerting.model
+
+/**
+ * This model is a wrapper for [Alert] that should only be used to create a more
+ * informative alert object to enrich mustache template notification messages.
+ */
+data class AlertContext(
+    val alert: Alert,
+    val associatedQueries: List<DocLevelQuery>? = null,
+    val sampleDocs: List<Map<String, Any?>>? = null
+) : Alert(
```

Comment:
Non-blocking

Inheritance seems like overkill here. The context could just have an associated alert as a member variable rather than extending an alert. 

Inheritance usually means that A is a B. In this context, I would argue an AlertContext is not an Alert. But I understand that using inheritance here limits the scope of the changes since AlertContext still evaluates to an Alert when typed variables are used.

Reply:
Yeah, I went with this approach of extending the Alert class because I essentially wanted to build upon the functionality of the Alert data model to make a specialized "Alert" that could be used in instances when we want additional information (in this case, when we want to expose more info in the notification message). I'll revise the PRs to make AlertContext a separate class.